apps/web/src/ee/clerk/providers/EnterpriseAuthProvider.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { createContext, useContext, useEffect, useCallback, useMemo, useState } from 'react';
+import { useLDClient } from 'launchdarkly-react-client-sdk';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { useQueryClient } from '@tanstack/react-query';
+import * as Sentry from '@sentry/react';
+import type { IOrganizationEntity, IUserEntity } from '@novu/shared';
+import { useAuth, useUser, useOrganization, useOrganizationList } from '@clerk/clerk-react';
+import { OrganizationResource, UserResource } from '@clerk/types';
+import { useSegment } from '../../../components/providers/SegmentProvider';
+import { PUBLIC_ROUTES_PREFIXES, ROUTES } from '../../../constants/routes';
+import { IS_CLERK_AUTH_ENABLED } from '../../../config/index';
+
+interface AuthEnterpriseContextProps {
+  inPublicRoute: boolean;
+  inPrivateRoute: boolean;
+  isLoading: boolean;
+  currentUser: IUserEntity | undefined;
+  organizations: IOrganizationEntity[];
+  currentOrganization: IOrganizationEntity | undefined;
+  login: (...args: any[]) => void;
+  logout: () => void;
+  environmentId: string | null;
+}
+
+const toUserEntity = (clerkUser: UserResource): IUserEntity => ({
+  _id: clerkUser.id,
+  firstName: clerkUser.firstName,
+  lastName: clerkUser.lastName,
+  email: clerkUser.emailAddresses[0].emailAddress,
+  profilePicture: clerkUser.imageUrl,
+  createdAt: clerkUser.createdAt?.toString() ?? '',
+  showOnBoarding: clerkUser.publicMetadata.showOnBoarding,
+  showOnBoardingTour: clerkUser.publicMetadata.showOnBoardingTour,
+  servicesHashes: clerkUser.publicMetadata.servicesHashes,
+  jobTitle: clerkUser.publicMetadata.jobTitle,
+  hasPassword: clerkUser.passwordEnabled,
+});
+
+const toOrganizationEntity = (clerkOrganization: OrganizationResource): IOrganizationEntity => ({
+  _id: clerkOrganization.id,
+  name: clerkOrganization.name,
+  createdAt: clerkOrganization.createdAt.toString(),
+  updatedAt: clerkOrganization.updatedAt.toString(),
+  apiServiceLevel: clerkOrganization.publicMetadata.apiServiceLevel,
+  defaultLocale: clerkOrganization.publicMetadata.defaultLocale,
+  domain: clerkOrganization.publicMetadata.domain,
+  productUseCases: clerkOrganization.publicMetadata.productUseCases,
+});
+
+const AuthEnterpriseContext = createContext<AuthEnterpriseContextProps | undefined>(undefined);
+
+export const useAuthEnterpriseContext = () => {
+  const context = useContext(AuthEnterpriseContext);
+  if (context === undefined) {
+    throw new Error('useAuthEnterpriseContext must be used within an AuthEnterpriseProvider');
+  }
+
+  return context;
+};
+
+const _AuthEnterpriseProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
```

Comment:
Technically this is the community auth provider. The enterprise is the Clerk one. So I'd wrap the application with an AuthProvider component or use the HOC I mentioned before depending if the app loads in EE or community edition.

Reply:
This is actually the enterprise one, the community one is `useAuthCommunity` hook - but I've moved the ClerkProvider inside this class so we have just one provider for EE.